presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -0,0 +1,1321 @@
+                                return null;
+                            })));
+        }
+        catch (AlreadyExistsException e) {
+            throw new SchemaAlreadyExistsException(database.getName());
+        }
+        catch (TException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public void dropDatabase(MetastoreContext metastoreContext, String databaseName)
+    {
+        try {
+            retry()
+                    .stopOn(NoSuchObjectException.class, InvalidOperationException.class)
+                    .stopOnIllegalExceptions()
+                    .run("dropDatabase", stats.getAlterDatabase().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> {
+                                client.dropDatabase(databaseName, false, false);
+                                return null;
+                            })));
+        }
+        catch (NoSuchObjectException e) {
+            throw new SchemaNotFoundException(databaseName);
+        }
+        catch (TException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public void alterDatabase(MetastoreContext metastoreContext, String databaseName, Database database)
+    {
+        try {
+            retry()
+                    .stopOn(NoSuchObjectException.class, MetaException.class)
+                    .stopOnIllegalExceptions()
+                    .run("alterDatabase", stats.getAlterDatabase().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> {
+                                client.alterDatabase(databaseName, database);
+                                return null;
+                            })));
+        }
+        catch (NoSuchObjectException e) {
+            throw new SchemaNotFoundException(databaseName);
+        }
+        catch (TException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    @Override
+    public void createTable(MetastoreContext metastoreContext, Table table)
+    {
+        try {
+            retry()
+                    .stopOn(AlreadyExistsException.class, InvalidObjectException.class, MetaException.class, NoSuchObjectException.class)
+                    .stopOnIllegalExceptions()
+                    .run("createTable", stats.getCreateTable().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> {
+                                client.createTable(table);
+                                return null;
+                            })));
+        }
+        catch (AlreadyExistsException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 818.